sys/random/mersenne.c

```diff
@@ -129,7 +133,12 @@ uint32_t genrand_uint32(void)
     y ^= (y << 7) & 0x9d2c5680UL;
     y ^= (y << 15) & 0xefc60000UL;
     y ^= y >> 18;
+
+#ifdef MODULE_ENTROPY
+    return y + entropy;
```

Comment:
> Apart from that, I think we have several levels of needed 'randomness'.
>
> * "bootup-safe" -> 1000(*n) nodes booting up at the same time will not have the same IP address >   if that is derived from our "random number generator"
>
> * "network-safe" -> "random" back-off will not collide between nodes
>
> * "crypto-safe" -> the higher gods of crypto math consider it "random" enough for safe crypto

I guess that's exactly why "the big OSs" have varying degrees of RNGs:
* `rand()` in libc for quick, but deterministic PRNG values
* `/dev/random` for slow, but true randomness from outside sources
* `/dev/urandom` for something in between

I saw the `random` module always as something like `rand()` and `periph/random` as our `/dev/random`. True, for boards without `periph/random` we still need a randomness source, and maybe this is a start, but I wouldn't touch the PRNGs for that (except that we still need some way to initialize them properly).